packages/incremental/src/context.candy

```diff
@@ -0,0 +1,37 @@
+use ..query
+
+class QueryContext {
+  /// A context that orchestrates running queries.
+
+  static fun create(): QueryContext {
+    QueryContext(
+      MutableMap.empty<String, MutableMap<Equals & Hash, Equals & Hash>>(),
+      MutableList.empty<(String, Equals & Hash)>()
+    )
+  }
+
+  let cache: MutableMap<String, MutableMap<Equals & Hash, Equals & Hash>>
+  /// A cache from query names to a query-specific cache that maps from input to output.
```

Comment:
What about `MutableMap<(String, Equals & Hash), Equals & Hash>` – wouldn't that be simpler?

Reply:
This won't work because `(Equals & Hash, Equals & Hash)` doesn't implement `Equals & Hash`.